content/webapp/components/EventPromo/EventPromo.tsx

```diff
@@ -23,10 +25,18 @@ type Props = {
   fromDate?: Date;
 };
 
-function getLocationText(isOnline?: boolean, place?: Place): string {
-  if (!isOnline && place) return place.title;
+function getLocationText(isOnline?: boolean, place?: Place[]): string {
```

Comment:
This is the sort of function where it'd be easy (and maybe useful) to have a bunch of unit tests.

I'd also add a comment explaining what the intended rules are, so it's easier to check they're correct.

Is it worth adding a special case for "Throughout the building", so those events display consistently with other events on the site?